app/controllers/declaration/AdditionalDeclarationTypePageController.scala

```diff
@@ -59,12 +57,13 @@ class AdditionalDeclarationTypePageController @Inject()(
       .form()
       .bindFromRequest()
       .fold(
-        (formWithErrors: Form[AdditionalDeclarationType]) =>
-          Future.successful(BadRequest(declaration_type(formWithErrors))),
-        validAdditionalDeclarationType =>
+        formWithErrors => Future.successful(BadRequest(declaration_type(formWithErrors))),
+        validAdditionalDeclarationType => {
+          updateCache(journeySessionId, validAdditionalDeclarationType)
```

Comment:
Doesn't look like the response (future) of `updateCache` is ever used. I appreciate this is intentional, we could maybe make it more obvious though for when we change over e.g.
```
for {
_ <- updateCache(...)
_ <- customsCacheService.cache(...).map(
} yield Redirect(...)
```

Reply:
This would mean we wait until *both*  futures have finished before sending the Redirect. At the moment we don't wait until the new `updateCache` finishes, so it could fail and we'd never know

Reply:
yep agreed, will make this change to the other controllers as well